cmp.c

```diff
@@ -118,7 +118,7 @@ const char * const cmp_error_messages[ERROR_MAX + 1] = {
   "Max Error"
 };
 
-#if WORDS_BIGENDIAN == 0
+#if !defined(WORDS_BIGENDIAN) || WORDS_BIGENDIAN == 0
```

Comment:
@camgunz I just read the readme, and this code doesn't agree with the readme, which says "if not defined, checks at runtime". That's not true. If it's not defined, it defaults to 0 in this comparison. The change I made here causes no semantic change. The code was wrong before and is equally wrong after.

What would you like to do here?

Reply:
Oh you're right! I was careless when I added in `WORDS_BIGENDIAN` here, good catch.

I was going for a compile-time override here, so yeah I think what I want is

```c
#ifdef WORDS_BIGENDIAN
#define is_bigendian() (WORDS_BIGENDIAN)
#else
static const int32_t _i = 1;
#define is_bigendian() ((*(char *)&_i) == 0)
#endif
```

Reply:
Ok, I've changed the code. Commit message:

This is a semantic change: previously, if `WORDS_BIGENDIAN` was not defined, the code would
statically default to little endian by assuming `WORDS_BIGENDIAN == 0`. With this change, if
`WORDS_BIGENDIAN` is not defined, the code will default to runtime detection, adding a memory
lookup (hopefully cached) and a branch (hopefully branch-predicted correctly).

This is the documented behaviour in README.md. To avoid performance regressions, users of the
library should always explicitly compile with `-DWORDS_BIGENDIAN=0`.

Reply:
Turns out, clang/gcc are smart enough to not do those runtime things. They statically compile out the check.